packages/unmock-cli/src/cli.ts

```diff
@@ -5,100 +5,102 @@ import fs from "fs";
       });
-      child.on("exit", () => {
-        logger.log(`Done editing ${hash}.`);
+      const { data } = await axios({
+        ...(options.data ? { data: JSON.parse(options.data) } : {}),
+        ...(options.headers ? { headers: makeHeaders(options.header) } : {}),
+        method: options.request || "GET",
+        url,
       });
-    } else {
-      logger.log(`Could not find ${hash}`);
-    }
-  });
+      process.stdout.write(JSON.stringify(data, null, 2) + "\n");
+    });
 
-program
-  .command("list")
-  .action(() => {
-    const logger = new WinstonLogger();
-    glob("__snapshots__/**/*.snap", (e, matches) => {
-      if (e) {
-        throw e;
+  program
+    .command("open <hash>")
+    .option("-e, --editor [editor]", "editor (defaults to the EDITOR env variable or, in the absence of this, vi)")
+    .action((hash, options) => {
+      const logger = new WinstonLogger();
+      const editor = options.editor || process.env.EDITOR || "vi";
+      const path = `${process.cwd()}/.unmock/save/${hash}/response.json`;
+      if (fs.existsSync(path)) {
+        const child = child_process.spawn(editor, [path], {
+            stdio: "inherit",
+        });
+        child.on("exit", () => {
+          logger.log(`Done editing ${hash}.`);
+        });
+      } else {
+        logger.log(`Could not find ${hash}`);
       }
-      matches.forEach((match) => {
-        logger.log(match.substring(0, match.length - 5).substring(14));
-        const snapshots = require(`${process.cwd()}/${match}`);
-        const tests = Object.keys(snapshots)
-          .map(name => ({[name.split(" ").slice(0, -1).join(" ")]: new Array<string>()}))
-          .reduce((a, b) => ({ ...a, ...b}), {});
-        Object.keys(snapshots)
-          .forEach(name => {
-            tests[name.split(" ").slice(0, -1).join(" ")].push(name);
-          });
-        Object.keys(tests)
-          .forEach(test => {
-            logger.log(`  ${test}`);
-            tests[test].forEach(call => {
-              logger.log(`    hash: ${snapshots[call].hash}`);
-              logger.log(`      method: ${snapshots[call].method}`);
-              logger.log(`      host: ${snapshots[call].host}`);
-              logger.log(`      path: ${snapshots[call].path}`);
+    });
+
+  program
+    .command("list")
+    .action(() => {
+      const logger = new WinstonLogger();
+      glob("__snapshots__/**/*.snap", (e, matches) => {
+        if (e) {
+          throw e;
+        }
+        matches.forEach((match) => {
+          logger.log(match.substring(0, match.length - 5).substring(14));
+          const snapshots = require(`${process.cwd()}/${match}`);
+          const tests = Object.keys(snapshots)
+            .map(name => ({[name.split(" ").slice(0, -1).join(" ")]: new Array<string>()}))
+            .reduce((a, b) => ({ ...a, ...b}), {});
+          Object.keys(snapshots)
+            .forEach(name => {
+              tests[name.split(" ").slice(0, -1).join(" ")].push(name);
```

Comment:
The `.split(" ").slice(0, -1).join(" ")` operations seems to repeat itself - should this be either a separate function or alternatively, one can iterate over `Object.keys(snapshots)` once and just build `tests` in one go?

Reply:
:+1: